### PR TITLE
add gitlab project avatar & display name. add gitlab group description

### PR DESCRIPTION
**Pull request by casz**

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java [hunk 1]
```diff
@@ -542,7 +542,7 @@ protected SCMRevision retrieve(@NonNull String thingName, @NonNull TaskListener
                 GitLabApi gitLabApi = apiBuilder(serverName);
                 listener.getLogger().format("Looking up project %s%n", projectPath);
                 gitlabProject = gitLabApi.getProjectApi().getProject(projectPath);
-                result.add(new ObjectMetadataAction(null, gitlabProject.getDescription(),
+                result.add(new ObjectMetadataAction(gitlabProject.getNameWithNamespace(), gitlabProject.getDescription(),
```
Comment (baymac) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java]: The display name is updated but the description of gitlab project doesn't seem to be updated. :/

Comment (casz) [reply src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java]: Maybe I miss understand the object metadata for owner and repo 😓

Comment (casz) [reply src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java]: I'll have to investigate 😓

Comment (casz) [reply src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java]: Does not work at all 😭

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java [hunk 2]
```diff
@@ -535,22 +536,24 @@ protected SCMRevision retrieve(@NonNull String thingName, @NonNull TaskListener
...
     @Override
     protected List<Action> retrieveActions(SCMSourceEvent event, @NonNull TaskListener listener) {
-        LOGGER.info(String.format("e, l..%s", Thread.currentThread().getName()));
         List<Action> result = new ArrayList<>();
         if (gitlabProject == null) {
             try {
                 GitLabApi gitLabApi = apiBuilder(serverName);
                 listener.getLogger().format("Looking up project %s%n", projectPath);
                 gitlabProject = gitLabApi.getProjectApi().getProject(projectPath);
-                result.add(new ObjectMetadataAction(null, gitlabProject.getDescription(),
```
Comment (casz) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java]: actions we only added once but as soon as navigated away the object metadata would be gone 😆

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/helpers/GitLabHelper.java [hunk 3]
```diff
@@ -20,7 +20,7 @@ public static GitLabApi apiBuilder(String serverName) throws NoSuchFieldExceptio
             }
             return new GitLabApi(server.getServerUrl(), GitLabServer.EMPTY_TOKEN);
         }
-        throw new NoSuchFieldException(
+        throw new IllegalStateException(
```
Comment (casz) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/helpers/GitLabHelper.java]: I think this is an improvement 😅

**Comment by casz** [comment]
So for project the display name is only available on mouse over?
Is this a bug in the SCM api or branch API? cc @stephenc 
image

**Comment by casz** [comment]
I wonder if SCMSourceBuilder's project name is the one being used?

**Comment by stephenc** [comment]
The branch name is what people recognise, so it is the name displayed. The Description is on hover and is set from the description metadata action

**Comment by casz** [comment]
@stephenc this is the repository list 😅

**Comment by stephenc** [comment]
Behaviour is the same... repository name is like branch name

**Comment by casz** [comment]
I guess is the best we are gonna get with GitLab.
@baymac Unless you want to change strategy from using the project path to remove the group name but still store it on the SCM source.

Like here
https://ci.jenkins.io/job/Plugins/

Though you can have subgroups with conflicting project names but then I would argue for using the following:
Omit group/user name on a SCM source then use `Project Name` or `Sub group name / Project name`

This is the code in question
https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/224bdba9df52e67f243bec9a72af4891314e4e4b/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSourceBuilder.java#L24

It calls super and the second attribute is what is used for repository list.

**Comment by baymac** [comment]
@casz I have also been testing, I am not sure if it is `SCMNavigatorRequest` or `SCMSourceBuilder` which is setting the display name of each project with the `pathWithNamespace`. Although I am fine with the current state, main part which is description of group and single project both works.

**Comment by casz** [comment]
I'll do a separate PR for the display name as we most likely will have to add migration path for anyone currently using the plugin :)

**Comment by casz** [comment]
For anyone in the future.

Naming on SCM API happens on [request.process](https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/d164e8e4a23560cc7e389af0898072b9a69a7a9a/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java#L264)

**Pull request merged**